src/Dfc.CourseDirectory.WebV2/Features/Courses/DeleteCourseRun.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.Core.Models;
+using Dfc.CourseDirectory.Core.Validation;
+using Dfc.CourseDirectory.WebV2.Behaviors;
+using FluentValidation.Results;
+using MediatR;
+using OneOf;
+using DeleteCourseRunQuery = Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries.DeleteCourseRun;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Courses.DeleteCourseRun
+{
+    public class Request : IRequest<ViewModel>
+    {
+        public Guid CourseId { get; set; }
+        public Guid CourseRunId { get; set; }
+    }
+
+    public class ViewModel : Command
```

Comment:
I prefer the one class per file rule. I'd probably use a subfolder/namespace to keep the pieces together. Obviously one for later as we'd need to consider the whole project.

Reply:
Yeh I think we need to have a separate PR to update everything if we're going to change it. Gives me chance to convince you that this way is better ;-)